filechooser-portal/Main.vala

```diff
@@ -343,10 +333,61 @@ public class Files.FileChooserPortal : Object {
         yield;
 
         dialogs.remove (parent_window);
+        dialog.destroy ();
+
         response = _response;
         results = _results;
     }
 
+    private async bool check_overwrite_uri (string? uri) {
+        if (uri == null) {
+            return false;
+        }
+
+        var file = GLib.File.new_for_uri (uri);
+        if (file.query_exists ()) {
+            var display_name = file.get_basename ();
+            var primary = _("Replace “%s”?").printf (display_name);
+            unowned var secondary = _("Replacing this file will overwrite its current contents");
+            if (file.query_file_type (FileQueryInfoFlags.NOFOLLOW_SYMLINKS) == FileType.SYMBOLIC_LINK) {
+                try {
+                    var info = file.query_info (FileAttribute.STANDARD_SYMLINK_TARGET, FileQueryInfoFlags.NONE);
+                    if (info != null) {
+                        primary = _("Replace “%s”?".printf (info.get_symlink_target ()));
+                    } else {
+                        primary = _("Replace the target of “%s”?".printf (display_name));
+                    }
+
+                    secondary = _("Replacing the target file for this link will overwrite its current contents.");
```

Comment:
Trying to workshop the wording for the link case. I think this is more clear to me, but open to input. Especially whether the copy at the end about "the link will remain" is necessary.
Note that this code fails to compile when `secondary` is unowned. I'm also not sure how to handle the very long line that it produces according to elementary style.

```suggestion
                    primary = _("Replace link “%s”?").printf (display_name);
                    if (info != null) {
                        secondary = _("This file is a link to “%s”. Replacing it will overwrite the original's contents. The link will remain.").printf (info.get_symlink_target ());
                    } else {
                        secondary = _("This file is a link. Replacing it will overwrite the original's contents. The link will remain.");
                    }
```

Also note that the original code uses the translation macro improperly. The `_()` call should end before `printf`. This should be fixed whether my new copy is used or not.

Reply:
Fixed

Reply:
Seems this is not fixed after all ...